src/components/Authentication/OAuth.vue

```diff
@@ -0,0 +1,58 @@
+<template lang="pug">
+v-container.oauth-widget
+  h4 Or {{ verb }} with OAuth
+  v-btn.no-decorate(v-for="provider in providers",
+      :key="provider.id",
+      :dark="iconMap[provider.id].dark",
+      :color="provider.id",
+      :href="provider.url")
+    v-icon.aligned(left) {{ $vuetify.icons[iconMap[provider.id].icon] }}
+    | {{ provider.name }}
+</template>
+
+<script>
+const iconMap = {
+  github: { dark: true, icon: 'github' },
+  google: { dark: true, icon: 'google' },
+  linkedin: { dark: true, icon: 'linkedin' },
+  bitbucket: { dark: false, icon: 'bitbucket' },
+  box: { dark: true, icon: 'box_com' },
+  globus: { dark: true, icon: 'globus' },
+};
+export default {
+  props: {
+    verb: {
+      type: String,
+      default: 'sign in',
+    },
+    // Array of Object{ id: String, name: String, url: String }
+    providers: {
+      type: Array,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      enabledProviders: [],
+      iconMap,
+    };
+  },
+};
+</script>
+
+<style lang="stylus" scoped>
+.no-decorate
+  text-decoration none
+
+.v-btn.google
+  background-color #3367d6 !important
+
+.v-btn.linkedin
+  background-color #283e4a !importan
+
+.v-btn.box
+  background-color #0071f7 !important
+
+.v-btn.globus
+  background-color #335a95 !important
```

Comment:
I think we should nest the above rules for the sake of DRY:

```stylus
.v-btn
  &.globus
    background-color #335a95 !important

  &.box
    background-color #0071f7 !important

  // etc
```

That way if for some reason we use something other than `.v-btn` as the containing element we would just have to change it in one line rather than several.